hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAsyncNonMetaRegionLocator.java

```diff
@@ -55,35 +58,53 @@
 import org.junit.ClassRule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Category({ MediumTests.class, ClientTests.class })
+@RunWith(Parameterized.class)
```

Comment:
Ah, hey, you made it work! Thank you. I don't know if it works better for you or not, but there's also `MiniClusterRule` in hbase-server.

Reply:
I roughly went through MiniClusterRule, seems that I still need to keep  @BeforeClass to do set of things. Would like to use it in the new tests, keep it as it is for existing test case for now.